staging/src/k8s.io/client-go/openapi/groupversion_test.go

```diff
@@ -0,0 +1,198 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openapi
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/kube-openapi/pkg/handler3"
+)
+
+func TestGroupVersionOpenAPIV3(t *testing.T) {
+	// create an instance of the server without starting it
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var objJSON []byte
+		switch r.URL.Path {
+		case "/openapi/v3/apis/apps/v1":
+			if r.URL.RawQuery != "hash=014fbff9a07c" {
+				w.WriteHeader(http.StatusNotFound)
+			}
+			objJSON = []byte(`{"openapi":"3.0.0","info":{"title":"Kubernetes","version":"unversioned"}}`)
+		default:
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write(objJSON)
+	}))
+	defer server.Close()
+
+	fakeOpenAPIV3DiscoveryGV := handler3.OpenAPIV3DiscoveryGroupVersion{ServerRelativeURL: "/openapi/v3/apis/apps/v1?hash=014fbff9a07c"}
+	c, err := rest.RESTClientFor(&rest.Config{
+		Host: server.URL,
+		ContentConfig: rest.ContentConfig{
+			NegotiatedSerializer: scheme.Codecs,
+			GroupVersion:         &appsv1.SchemeGroupVersion,
+		},
+	})
+	if err != nil {
+		t.Fatalf("unexpected error occurred: %v", err)
+	}
+	gv := newGroupVersion(&client{restClient: c}, fakeOpenAPIV3DiscoveryGV, true)
```

Comment:
rather than constructing fakeOpenAPIV3DiscoveryGV and replicating the calls to newGroupVersion, actually exercise the client logic (same comment applies to other instances):

```go
openapiClient := NewClient(c)
paths, err := openapiClient.Paths()
if err != nil {
  t.Fatal(err)
}
schema, err := paths["apps/v1"].Schema(runtime.ContentTypeJSON)
if err != nil {
  t.Fatal(err)
}
...
```